reposurgeon: add --with-cython option

dlh opened the pull request:
This option builds and installs a `cyreposurgeon` executable.

There are two commits in this pull request, but feel free to squash into one. The python requirement for Snow Leopard is not related to the added `--with-cython` option.

On my machine it took about 50 minutes for clang to compile the 18 MB `.c` file that cython generates, so I added an `ohai` notice stating that it may take a while.

dunn commented:
I wonder if the warning about build time should be added to the option description instead?

mistydemeo commented:
Is there a disadvantage to enabling this by default?

DomT4 commented on Library/Formula/reposurgeon.rb:
```diff
@@ -27,6 +35,18 @@ def install
+        system "python", *Language::Python.setup_install_args(buildpath/"vendor")
+      end
+      ENV.prepend_path "PYTHONPATH", buildpath/"vendor/lib/python2.7/site-packages"
+      ohai "Building cyreposurgeon", "This may take a while"
```
Let's not `ohai`. People can flip the `--verbose` switch if they worry about builds appearing to stall.

dlh commented:
Updated. Removed the `ohai`.

@mistydemeo: the only disadvantage is the build time & clang memory usage, but that issue would be mitigated since almost all users would be installing a bottle. If you're okay with Brew Test Bot handling the long build then I'll enable it by default.

mikemcquaid commented:
Yeh, let's try it by default, thanks!

dlh commented:
Updated. It's on by default now.

mikemcquaid commented:
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!
